cibuildwheel/util.py

```diff
@@ -61,11 +63,25 @@ class IdentifierSelector:
     build identifier, and it returns True if that identifier should be
     included.
     """
-    def __init__(self, *, build_config: str, skip_config: str):
+    def __init__(self, *, build_config: str, skip_config: str, requires_python: Optional[SpecifierSet] = None):
         self.build_patterns = build_config.split()
         self.skip_patterns = skip_config.split()
+        self.requires_python = requires_python
 
     def __call__(self, build_id: str) -> bool:
+        # Filter build selectors by python_requires if set
+        if self.requires_python is not None and '-' in build_id:
```

Comment:
Shouldn't we assert for `'-' in build_id`?

Reply:
We haven't in the past, and there are unit tests that put really weird things in here. That's why this is here. :)

Reply:
Aaaaaaaaah, right. Forgot about the unit tests again :-/
Hmmm, that's annoying, changing our code's assumptions to fit unit tests, but... changing unit tests to work around this is almost just as fishy.

Reply:
mmmyeah, this does seem a little wrong... also the ValueError catch below. These shouldn't be hit in real use, so I think we should fix the unit tests. It looks like only `test_build_selector` in `main_options_test.py`?